### PR TITLE
InRamChunkedMmap: optimize using madvise

### PR DESCRIPTION
**Pull request by xzfc**
Follow-up for #4828.

### CODE REVIEW

File: lib/segment/src/vector_storage/chunked_mmap_vectors.rs [hunk 1]
```diff
@@ -221,6 +225,42 @@ impl<T: Sized + Copy + 'static> ChunkedMmapVectors<T> {
...
+        let mut it = 0..self.len();
+
+        for (chunk_idx, chunk) in self.chunks.iter().enumerate() {
+            for (i, _) in zip(0..self.config.chunk_size_vectors, &mut it) {
+                let idx = i * self.config.dim;
+                f(&chunk[idx..idx + self.config.dim])?;
+            }
+
+            chunk.flusher()()?;
+            fn _assert_t_is_copy<T: Copy>() {}
```
Comment (timvisee) [inline lib/segment/src/vector_storage/chunked_mmap_vectors.rs:242]: What is this line for?

File: lib/sparse/src/index/inverted_index/inverted_index_mmap.rs [hunk 2]
```diff
@@ -196,8 +195,7 @@ impl InvertedIndexMmap {
         let file_header: InvertedIndexFileHeader = read_json(&config_file_path)?;
         // read index data into mmap
         let file_path = Self::index_file_path(path.as_ref());
-        let mmap = open_read_mmap(file_path.as_ref())?;
-        madvise::madvise(&mmap, madvise::Advice::Normal)?;
+        let mmap = open_read_mmap(file_path.as_ref(), Advice::Normal.into())?;
```
Comment (generall) [inline lib/sparse/src/index/inverted_index/inverted_index_mmap.rs:198]: For the sake of better code navigation, let's use something like this:

```suggestion
        let mmap = open_read_mmap(file_path.as_ref(), AdviceSetting::from(Advice::Normal))?;
```

**Review by generall — APPROVED** [review]
LGTM, with minor nitpick.

File: lib/segment/src/vector_storage/chunked_mmap_vectors.rs (hunk 1 above)
Comment (timvisee) [reply lib/segment/src/vector_storage/chunked_mmap_vectors.rs]: I see now. Interesting hack to enforce the `Copy` trait :smile:

**Review by timvisee — APPROVED** [review]

**Pull request merged**